### ISSUE TITLE
Confused with IP mappings.

### ISSUE THREAD

**Issue by segg21** [eclipse-che/che#1303]
Ok I understand theres multiple network adaptors, and ceh runs on the ip of 192.168.99.100 and my vps running on 192.168.0.129.

My question is, what ip should I port forward too? I had it working randomly, but now its not working anymore after I reinstall and installed everything back the right way. I have no clue what I did to get it working the first time but can I get some personal assistance?

The weird thing is when I checked my adapter properties the ip of the adapter was 192.168.99.1 but I connect through 192.168.99.100. Im soo confused

I tried re-routing my ip 192.168.0.129 to 192.168.9.1 then just portforward my computer like normal but that didnt even work

**Comment by segg21** [comment]
Ok, what I did was manually deleted all my adapters, and uninstalled virtual box, and everything it had. Restarted, and port forwarded my vps ip/port and now its visible. 

I still need more person assistance because this is something new to me. I'm going to be using github, and I have setup the auth and what not, but how do i disable the dashboard so only people who logged in has there own dashboard and ability to create there own projects?

**Comment by segg21** [comment]
Cannot connect to che.. I applied my network information which was ip 192.168.99.1 for che. I still get the timeout error even after increasing the ms time, and after it expired the error of 

```
[STDOUT] Listening for transport dt_socket at address: 4403
[STDOUT] 2016-05-20 02:22:31,666[main] [INFO] [o.a.c.s.VersionLoggerListener 89] - Server version: Apache Tomcat/8.0.32
[STDOUT] 2016-05-20 02:22:31,691[main] [INFO] [o.a.c.s.VersionLoggerListener 91] - Server built: Feb 2 2016 19:34:53 UTC
[STDOUT] 2016-05-20 02:22:31,696[main] [INFO] [o.a.c.s.VersionLoggerListener 93] - Server number: 8.0.32.0
[STDOUT] 2016-05-20 02:22:31,709[main] [INFO] [o.a.c.s.VersionLoggerListener 95] - OS Name: Linux
[STDOUT] 2016-05-20 02:22:31,724[main] [INFO] [o.a.c.s.VersionLoggerListener 97] - OS Version: 4.4.8-boot2docker
[STDOUT] 2016-05-20 02:22:31,728[main] [INFO] [o.a.c.s.VersionLoggerListener 99] - Architecture: amd64
[STDOUT] 2016-05-20 02:22:31,744[main] [INFO] [o.a.c.s.VersionLoggerListener 101] - Java Home: /opt/jre1.8.0_65
[STDOUT] 2016-05-20 02:22:31,794[main] [INFO] [o.a.c.s.VersionLoggerListener 103] - JVM Version: 1.8.0_65-b17
[STDOUT] 2016-05-20 02:22:31,804[main] [INFO] [o.a.c.s.VersionLoggerListener 105] - JVM Vendor: Oracle Corporation
[STDOUT] 2016-05-20 02:22:31,810[main] [INFO] [o.a.c.s.VersionLoggerListener 107] - CATALINA_BASE: /home/user/che/ws-agent
[STDOUT] 2016-05-20 02:22:31,823[main] [INFO] [o.a.c.s.VersionLoggerListener 109] - CATALINA_HOME: /home/user/che/ws-agent
[STDOUT] 2016-05-20 02:22:31,850[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Djava.util.logging.config.file=/home/user/che/ws-agent/conf/logging.properties
[STDOUT] 2016-05-20 02:22:31,872[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
[STDOUT] 2016-05-20 02:22:31,901[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Xms256m
[STDOUT] 2016-05-20 02:22:31,916[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Xmx1024m
[STDOUT] 2016-05-20 02:22:31,923[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Djava.security.egd=file:/dev/./urandom
[STDOUT] 2016-05-20 02:22:31,941[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dche.logs.dir=/home/user/che/ws-agent/logs
[STDOUT] 2016-05-20 02:22:31,948[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -agentlib:jdwp=transport=dt_socket,address=4403,server=y,suspend=n
[STDOUT] 2016-05-20 02:22:31,958[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dcom.sun.management.jmxremote
[STDOUT] 2016-05-20 02:22:31,966[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dcom.sun.management.jmxremote.ssl=false
[STDOUT] 2016-05-20 02:22:31,972[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dcom.sun.management.jmxremote.authenticate=false
[STDOUT] 2016-05-20 02:22:31,995[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dche.local.conf.dir=/mnt/che/conf
[STDOUT] 2016-05-20 02:22:32,005[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Djava.endorsed.dirs=/home/user/che/ws-agent/endorsed
[STDOUT] 2016-05-20 02:22:32,017[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dcatalina.base=/home/user/che/ws-agent
[STDOUT] 2016-05-20 02:22:32,031[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dcatalina.home=/home/user/che/ws-agent
[STDOUT] 2016-05-20 02:22:32,045[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Djava.io.tmpdir=/home/user/che/ws-agent/temp
[STDOUT] 2016-05-20 02:22:33,233[main] [INFO] [o.a.c.http11.Http11NioProtocol 425] - Initializing ProtocolHandler ["http-nio-4401"]
[STDOUT] 2016-05-20 02:22:33,308[main] [INFO] [o.a.t.util.net.NioSelectorPool 72] - Using a shared selector for servlet write/read
[STDOUT] 2016-05-20 02:22:33,329[main] [INFO] [o.a.catalina.startup.Catalina 591] - Initialization processed in 4949 ms
[STDOUT] 2016-05-20 02:22:33,523[main] [INFO] [c.m.JmxRemoteLifecycleListener 332] - The JMX Remote Listener has configured the registry on port 32002 and the server on port 32102 for the Platform server
[STDOUT] 2016-05-20 02:22:33,528[main] [INFO] [o.a.c.core.StandardService 435] - Starting service Catalina
[STDOUT] 2016-05-20 02:22:33,533[main] [INFO] [o.a.c.core.StandardEngine 259] - Starting Servlet Engine: Apache Tomcat/8.0.32
[STDOUT] 2016-05-20 02:22:34,348[ost-startStop-1] [INFO] [o.a.c.startup.HostConfig 910] - Deploying web application archive /home/user/che/ws-agent/webapps/ide.war
[STDOUT] Fri May 20 02:23:24 UTC 2016 - [localhost-startStop-1] Product-specified preferences called before plugin is started
[STDOUT] 2016-05-20 02:23:26,384[rrest.WSClient1] [INFO] [.c.a.c.n.WSocketEventBusClient 223] - Open connection to ws://che-host:8080/ide/api/eventbus/?token=dummy_token. 
[STDOUT] 2016-05-20 02:23:29,309[ost-startStop-1] [INFO] [o.a.c.startup.HostConfig 974] - Deployment of web application archive /home/user/che/ws-agent/webapps/ide.war has finished in 54,926 ms
[STDOUT] 2016-05-20 02:23:29,330[ost-startStop-1] [INFO] [o.a.c.startup.HostConfig 1030] - Deploying web application directory /home/user/che/ws-agent/webapps/ROOT
[STDOUT] 2016-05-20 02:23:29,970[ost-startStop-1] [INFO] [o.a.c.startup.HostConfig 1142] - Deployment of web application directory /home/user/che/ws-agent/webapps/ROOT has finished in 640 ms
[STDOUT] 2016-05-20 02:23:30,041[main] [INFO] [o.a.c.http11.Http11NioProtocol 470] - Starting ProtocolHandler ["http-nio-4401"]
[STDOUT] 2016-05-20 02:23:30,127[main] [INFO] [o.a.catalina.startup.Catalina 642] - Server startup in 56794 ms
[ERROR] Timeout reached. The Che server has been unable to verify that your workspace's agent has successfully booted. Either the workspace is unreachable, the agent had an error during startup, or your workspace is starting slowly. You can configure machine.ws_agent.max_start_time_ms in Che properties to increase the timeout.
```

**Comment by segg21** [comment]
Someone should make a tutorial because I cannot find no help at all with applying this with a network. Not even on youtube I can find a tutorial :/

**Comment by segg21** [comment]
Now im getting another error :(

```
java.net.BindException: Address already in use: JVM_Bind
        at java.net.DualStackPlainSocketImpl.bind0(Native Method) ~[na:1.8.0_91]
        at java.net.DualStackPlainSocketImpl.socketBind(Unknown Source) ~[na:1.8.0_91]
        at java.net.AbstractPlainSocketImpl.bind(Unknown Source) ~[na:1.8.0_91]
        at java.net.PlainSocketImpl.bind(Unknown Source) ~[na:1.8.0_91]
        at java.net.ServerSocket.bind(Unknown Source) ~[na:1.8.0_91]
        at java.net.ServerSocket.<init>(Unknown Source) ~[na:1.8.0_91]
        at org.apache.catalina.core.StandardServer.await(StandardServer.java:420) ~[catalina.jar:8.0.32]
        at org.apache.catalina.startup.Catalina.await(Catalina.java:717) [catalina.jar:8.0.32]
        at org.apache.catalina.startup.Catalina.start(Catalina.java:663) [catalina.jar:8.0.32]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_91]
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_91]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_91]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[na:1.8.0_91]
        at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:351) [bootstrap.jar:8.0.32]
        at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:485) [bootstrap.jar:8.0.32]
2016-05-19 21:02:55,566[main]             [INFO ] [o.a.c.http11.Http11NioProtocol 485]  - Pausing ProtocolHandler ["http-nio-8080"]
2016-05-19 21:02:55,568[main]             [INFO ] [o.a.c.core.StandardService 496]      - Stopping service Catalina
2016-05-19 21:02:55,777[ost-startStop-1]  [INFO ] [o.e.c.a.core.util.FileCleaner 74]    - File cleaner is stopped
2016-05-19 21:02:55,790[ost-startStop-1]  [WARN ] [o.e.c.a.m.s.MachineManager 1044]     - Failed delete C:\Users\LEGITS~1\AppData\Local\Programs\EclipseChe\eclipse-che-4.2.0\tomcat\logs\machine\logs\machineioj0z2eidtmlfcgl
2016-05-19 21:02:55,857[main]             [INFO ] [o.a.c.http11.Http11NioProtocol 514]  - Stopping ProtocolHandler ["http-nio-8080"]
2016-05-19 21:02:55,859[main]             [INFO ] [o.a.c.http11.Http11NioProtocol 529]  - Destroying ProtocolHandler ["http-nio-8080"]
```

**Comment by ghost** [comment]
`Address already in use` says it all. Makes sure you do not have another Che instance running by accident.

> I applied my network information which was ip 192.168.99.1 for che.

Can you elaborate here? Did you change anything in configuration? `192.168.99.1` is IP that ws agent uses to reach ws-master. And `192.168.99.100` is IP that ws-master uses to reach ws-agent. And according to the log, it's ws-master that cannot reach ws-agent not vice versa.

**Comment by ghost** [comment]
As to help articles, you will find info on Che networking here https://eclipse-che.readme.io/docs/networking

**Comment by segg21** [comment]
@eivantsov I removed the ip, and terminated all VM processes, and Che process and started. I still got the could not bind jvm. 

I've looked at the docs network but I'm still confused over the ip's. I currently have a domain/subdominant going to my main computer local IP address which I can connect with from port 8080. Basically anyone could visit so that's good.

Now with Che. When connecting through the domain, what ip should I use in Che? 192.268.99.100 correct? I did that but got the binding error of the ip already being in use. I don't know if I'm doing it correctly but it seems if I am
